### PR TITLE
sst_importer: impl SuspendImport interface

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/15611

What's Changed:
As title, the new RPC DenyImport has been added.

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/import/sst_service.rs [hunk 1]
```diff
@@ -638,6 +688,10 @@ macro_rules! impl_write {
 
             let timer = Instant::now_coarse();
             let label = stringify!($fn);
+            if let Err(err) = self.check_deny() {
```
Comment (glorv) [inline src/import/sst_service.rs:691]: What about only deny `ingest` and allow write/download since these request have no bad side-effect. In general, ingest is very fast so the retry cost is low but write&download is slow.

File: src/import/sst_service.rs [hunk 2]
```diff
@@ -1240,6 +1307,27 @@ impl<E: Engine> ImportSst for ImportSstService<E> {
...
+        ctx: RpcContext<'_>,
+        req: SuspendImportRpcRequest,
+        sink: UnarySink<SuspendImportRpcResponse>,
+    ) {
+        let label = "suspend_import_rpc";
+        let timer = Instant::now_coarse();
+
+        let suspended = if req.should_suspend_imports {
+            info!("suspend incoming import RPCs."; "for_second" => req.get_duration_in_secs(), "caller" => req.get_caller());
+            self.suspend_requests(Duration::from_secs(req.get_duration_in_secs()))
```
Comment (3pointer) [inline src/import/sst_service.rs:1322]: I think we need a limitation check for `req.get_duration_in_secs()`. for example 30mins.

File: src/import/sst_service.rs [hunk 3]
```diff
@@ -1240,6 +1307,27 @@ impl<E: Engine> ImportSst for ImportSstService<E> {
...
+        } else {
+            info!("allow incoming import RPCs."; "caller" => req.get_caller());
+            self.allow_requests()
+        };
+        let mut resp = SuspendImportRpcResponse::default();
+        resp.set_already_suspended(suspended);
```
Comment (3pointer) [inline src/import/sst_service.rs:1328]: why return previous suspended result here?

File: src/import/sst_service.rs [hunk 4]
```diff
@@ -1240,6 +1307,27 @@ impl<E: Engine> ImportSst for ImportSstService<E> {
         RawChunk,
         new_raw_writer
     );
+
+    fn suspend_import_rpc(
```
Comment (3pointer) [inline src/import/sst_service.rs:1311]: prefer name it `suspend_import`

File: src/import/sst_service.rs [hunk 5]
```diff
@@ -1240,6 +1306,27 @@ impl<E: Engine> ImportSst for ImportSstService<E> {
         RawChunk,
         new_raw_writer
     );
+
+    fn deny_import_rpc(
```
[the remark this answers is not in the document]
Comment (YuJuncen) [reply src/import/sst_service.rs]: There is a parameter in the request to do that.

File: src/import/sst_service.rs (hunk 1 above)
Comment (YuJuncen) [reply src/import/sst_service.rs]: Perhaps we can have a try.

File: src/import/sst_service.rs (hunk 3 above)
Comment (YuJuncen) [reply src/import/sst_service.rs:1328]: The client need this to check whether all the time importing has been suspened.

File: src/import/sst_service.rs (hunk 4 above)
Comment (YuJuncen) [reply src/import/sst_service.rs:1311]: Changing a RPC name needs to change all the repos we involved. We have just rename all `deny` to `suspend`. For now I think we'd better delay all renaming stuff to all behavior has been defined and we are about to merge all 3 prs.

File: src/import/sst_service.rs (hunk 2 above)
Comment (YuJuncen) [reply src/import/sst_service.rs:1322]: It is reasonable for suspend for a long time. (for example, longer ttl usually allow us to disconnect to TiKV for a longer time) I think if we need to check, the max should be at least about 5 ~ 6 hours.

File: src/import/sst_service.rs (hunk 3 above)
Comment (3pointer) [reply src/import/sst_service.rs:1333]: So the first time suspend request will return false to client?

Comment (YuJuncen) [reply src/import/sst_service.rs:1333]: Yes, and EXCEPT this time, each time we request suspending importing, we will check the return value.

File: Cargo.toml [hunk 6]
```diff
@@ -201,8 +201,8 @@ procinfo = { git = "https://github.com/tikv/procinfo-rs", rev = "7693954bd1dd86e
 # When you modify TiKV cooperatively with kvproto, this will be useful to submit the PR to TiKV and the PR to
 # kvproto at the same time.
 # After the PR to kvproto is merged, remember to comment this out and run `cargo update -p kvproto`.
-# [patch.'https://github.com/pingcap/kvproto']
-# kvproto = { git = "https://github.com/your_github_id/kvproto", branch = "your_branch" }
+[patch.'https://github.com/pingcap/kvproto']
+kvproto = { git = "https://github.com/yujuncen/kvproto", branch = "deny-import" }
```
Comment (BornChanger) [inline Cargo.toml:205]: please revert this change here since https://github.com/pingcap/kvproto/pull/1185 is merged.

File: components/sst_importer/src/metrics.rs [hunk 7]
```diff
@@ -127,4 +127,8 @@ lazy_static! {
         &["type"],
         exponential_buckets(0.01, 4.0, 8).unwrap()
     ).unwrap();
+    pub static ref IMPORT_DENIED_REQUESTS: IntCounter = register_int_counter!(
```
Comment (glorv) [inline components/sst_importer/src/metrics.rs:130]: What about reuse `IMPORTER_ERROR_VEC` since deny is expected to happen very rarely.

**Review by BornChanger — COMMENTED** [review]
Please update the title 

image

**Review by BornChanger — APPROVED** [review]
rest lgtm

**Review by 3pointer — APPROVED** [review]

**Review by pingyu — APPROVED** [review]
